components/Auth/Auth.js

```diff
@@ -0,0 +1,184 @@
+import { Component } from "react";
+import PropTypes from "prop-types";
+import axios from "axios";
+import get from "lodash/get";
+import { toast } from "react-toastify";
+
+import AuthContext from "../../contexts/AuthContext";
+
+class Auth extends Component {
+  static defaultProps = {
+    children: null,
+  };
+
+  static propTypes = {
+    children: PropTypes.node,
+  };
+
+  state = {
+    authToken: "",
+    user: null,
+    verifyingAuth: true,
+    loadingUserInfo: false,
+  };
+
+  api = axios.create({
+    baseURL: process.env.NEXT_PUBLIC_API_BASE_URL,
+  });
+
+  componentDidMount() {
+    try {
+      const authToken = localStorage.getItem("authToken");
```

Comment:
Keeping it in local storage not cookies?

Reply:
For now yes, I'll have to do some proxy stuff and make changes on the api. So thought I might do it later

Reply:
ok